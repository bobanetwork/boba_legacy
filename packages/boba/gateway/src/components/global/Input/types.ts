import { ReactNode } from 'react'

export interface InputProps {
  placeholder: string
  type: string
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
  value: any
  label?: ReactNode
}
