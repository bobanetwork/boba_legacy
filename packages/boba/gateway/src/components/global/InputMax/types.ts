export interface MaxInputTypes {
  max: number
  initialValue?: number
  onValueChange: (value: number) => void
}
