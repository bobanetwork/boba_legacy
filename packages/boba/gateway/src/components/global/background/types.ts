export enum BackgroundPosition {
  TOP = 'top',
  CENTER = 'center',
}

export interface BackgroundProps {
  position: BackgroundPosition
}

export const RoutesWithBackgroundPositionAtTop = ['/bridge', '/']
