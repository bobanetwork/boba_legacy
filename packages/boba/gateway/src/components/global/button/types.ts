import { ReactNode } from 'react'

type ButtonType = {
  disable?: boolean
  loading?: boolean
  small?: boolean
  outline?: boolean
  tiny?: boolean
  transparent?: boolean
}

export interface ButtonTypes extends ButtonType {
  label: ReactNode | string
  style?: Record<string, string>
  className?: string
  onClick?: () => void
}
