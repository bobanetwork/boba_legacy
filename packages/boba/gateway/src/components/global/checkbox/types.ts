export interface CheckboxWithLabelProps {
  label?: string
  checked?: boolean
  onChange: (isChecked: boolean) => void
}
