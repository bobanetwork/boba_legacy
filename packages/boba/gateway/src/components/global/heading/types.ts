import { ReactNode } from 'react'

export type VariantType = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'

export interface HeadingProps {
  variant: VariantType
  color?: string
  children?: ReactNode
  className?: string
}

export interface HeadingStyleProps {
  size: string
  lineHeight: string
  fontWeight: number
}
