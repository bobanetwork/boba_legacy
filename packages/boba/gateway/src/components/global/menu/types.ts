import { ComponentType, ReactElement, ReactNode } from 'react'

export interface OptionProps {
  label?: string
  component?: ReactElement<any>
  onClick?: (d: any) => void
}

export interface MenuProps {
  name?: string
  label?: ReactNode | string
  children?: ReactNode
  options: Array<OptionProps>
}
