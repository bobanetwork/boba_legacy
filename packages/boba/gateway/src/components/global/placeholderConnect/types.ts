export interface PlaceholderConnectInterface {
  preloader?: any
  isLoading?: boolean
}
