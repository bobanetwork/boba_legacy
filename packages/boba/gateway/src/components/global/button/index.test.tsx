import React from 'react'
import { render } from '@testing-library/react'
import { ButtonTypes } from './types'
import CustomThemeProvider from 'themes'
import { Button } from './'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

const mockStore = configureStore()

const renderButton = (props: ButtonTypes) => {
  return render(
    <Provider
      store={mockStore({
        ui: {
          theme: 'light',
        },
      })}
    >
      <CustomThemeProvider>
        <Button {...props}>Test Button</Button>
      </CustomThemeProvider>
    </Provider>
  )
}

describe('Button', () => {
  test('Default Button', () => {
    const { asFragment } = renderButton({
      label: 'Test Button',
    })
    expect(asFragment()).toMatchSnapshot()
  })
  test('Loading Button', () => {
    const { asFragment } = renderButton({
      loading: true,
      label: 'Test Button',
    })
    expect(asFragment()).toMatchSnapshot()
  })
  test('Disable Button', () => {
    const { asFragment } = renderButton({
      disable: true,
      label: 'Test Button',
    })
    expect(asFragment()).toMatchSnapshot()
  })
  test('Small Button', () => {
    const { asFragment } = renderButton({
      small: true,
      label: 'Test Button',
    })
    expect(asFragment()).toMatchSnapshot()
  })
  test('Outline Button', () => {
    const { asFragment } = renderButton({
      outline: true,
      label: 'Outline Button',
    })
    expect(asFragment()).toMatchSnapshot()
  })
})
