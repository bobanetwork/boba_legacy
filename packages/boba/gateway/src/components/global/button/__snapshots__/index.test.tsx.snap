// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Default Button 1`] = `
<DocumentFragment>
  <button
    class="sc-bgqQcB iCiqEp"
    label="Test Button"
    type="button"
  >
     Test Button
  </button>
</DocumentFragment>
`;

exports[`Button Disable Button 1`] = `
<DocumentFragment>
  <button
    class="sc-bgqQcB buwCqV"
    label="Test Button"
    type="button"
  >
     Test Button
  </button>
</DocumentFragment>
`;

exports[`Button Loading Button 1`] = `
<DocumentFragment>
  <button
    class="sc-bgqQcB kNhvec"
    label="Test Button"
    type="button"
  >
    <span
      class="sc-gTRrQi daKBPn"
    />
     Test Button
  </button>
</DocumentFragment>
`;

exports[`Button Outline Button 1`] = `
<DocumentFragment>
  <button
    class="sc-bgqQcB eGWGxW"
    label="Outline Button"
    type="button"
  >
     Outline Button
  </button>
</DocumentFragment>
`;

exports[`Button Small Button 1`] = `
<DocumentFragment>
  <button
    class="sc-bgqQcB hCCYhJ"
    label="Test Button"
    type="button"
  >
     Test Button
  </button>
</DocumentFragment>
`;
